internal/services/cognitive/cognitive_deployment_resource.go

```diff
@@ -315,6 +314,10 @@ func (r CognitiveDeploymentResource) Delete() sdk.ResourceFunc {
 		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
 			client := metadata.Client.Cognitive.DeploymentsClient
 
+			accountId := cognitiveservicesaccounts.NewAccountID(id.SubscriptionId, id.ResourceGroupName, id.AccountName)
```

Comment:
This is causing a build failure at the moment because `id` doesn't exist yet, we need to parse the deployment Id first before we can construct the account Id
```suggestion
			id, err := deployments.ParseDeploymentID(metadata.ResourceData.Id())
			if err != nil {
				return err
			}
			accountId := cognitiveservicesaccounts.NewAccountID(id.SubscriptionId, id.ResourceGroupName, id.AccountName)
```

Reply:
Thanks!  that's what i get for just blindly accepting suggestions :). Should be all good to go now